lib/install/management/delete.go

```diff
@@ -238,11 +238,33 @@ func (d *Dispatcher) DeleteVCHInstances(vmm *vm.VirtualMachine, conf *config.Vir
 	var err error
 	var children []*vm.VirtualMachine
 
-	// TODO: GET CHILDREN EITHER BY FOLDER OR RP BASED ON DEPLOYMENT
-	if children, err = d.parentResourcepool.GetChildrenVMs(d.op, d.session); err != nil {
+	parentFolder, err := vmm.Folder(d.op, d.session)
+	if err != nil {
 		return err
 	}
 
+	if parentFolder.Reference() == d.session.VMFolder.Reference() {
+		// use the resource pool to cut down on the number of candidates
+		d.op.Debugf("looking in the resource pool for delete targets")
+		if children, err = d.parentResourcepool.GetChildrenVMs(d.op, d.session); err != nil {
+			return err
+		}
+	} else {
+		// vch parent inventory folder exists, get the children from it.
+		folderChildren, err := parentFolder.Children(d.op)
```

Comment:
since we don't disable moving between folders we _may_ need to verify that the folder name is the same as the VCH -- that's not full proof either, but it would maintain parity w/resource pools...I'm doing some testing on that...if you have thoughts please share..

Reply:
disregard that...has nothing to do w/the name..we store the RP `moref` in the VCH config...which has it's on set of problems..but anyhoo...

Reply:
Understood, I was thinking about this as well. We may need to disable moving the vch and cvm's out of the the folder in the future for now. In addition to his, the vch can be moved and the folder reference will not be updated. So the cvms will still be made in the folder that is pointed to by the session. Once the vch is restarted it will change. But this means that we will most definitely leak cvms in this case. And it may not be easy to tell where they are, since we do not keep that folder around the best we could say is "you moved the vch(we think?) and so you have extra cvms around in the inventory." This would really only be the case for rob, but this also means we may need to check the rp after clearing out the inventory folder to make sure we have cleaned up all of the cvms. Or alteast let the user know that the rp still contains some cvms that we have not cleaned up.